src/painter.rs

```diff
@@ -0,0 +1,193 @@
+use crate::{
+    prompt::{PromptEditMode, PromptHistorySearch},
+    Prompt,
+};
+use crossterm::{
+    cursor::{self, position, MoveTo, MoveToColumn, RestorePosition, SavePosition},
+    style::{Color, Print, ResetColor, SetForegroundColor},
+    terminal::{self, Clear, ClearType},
+    QueueableCommand, Result,
+};
+
+use std::io::{Stdout, Write};
+
+pub struct Painter {
+    // Stdout
+    stdout: Stdout,
+}
+
+impl Painter {
```

Comment:
Most functions queue up the operations. Some also flush. Which makes the interface a bit inconsistent but satisfy the requirements more closely. What's other's take on 
- completely queue-based interface?
- flush after every function
- or a combination (proper naming)

Reply:
We'll probably need to do some measurements to see how bad flushing a lot is. My instinct is to say that we should only flush when we have to, so before reading an input (including terminal size), and not flush between normal writes to the screen. Probably need to play with a bit to find what feels best.